### PR TITLE
Removes /datum/Del, Del Profiling, Dels, and Unused Files That Contain Dels

### PR DESCRIPTION
**Pull request by Krausus**
So, turns out, `/datum/Del` adds a measurable amount of overhead to the deletion of any datum *by merely existing.* So, this PR gets rid of it. This has three main consequences:
- Deleting datums is faster, which is most noticeable for the ones that get created *a lot,* like matrices. Some things will probably lag noticeably less.
- General del profiling is no longer possible. The del profiling verbs have been removed.
  - Hard deletions are still tracked by the GC, and the verb to see those types is still present.
- Anything that gets directly deleted, either by `del` or BYOND's built-in GC, won't call its `Destroy` proc. With nothing using `del` to delete anything but clients, this could only cause issues with datums that happen to end up completely unreferenced. There probably aren't any of those that do anything important in their `Destroy` procs.

Since we don't really want anything (except clients) to be deleted through `del` anymore, I did a search for uses of it in the code. Instances of it have been replaced as follows:
- Stuff that needs to hard del? Replaced by qdel and a Destroy proc that returns the "hard del now" hint.
- Code that's commented out? Removed.
- Code in unused files? Removed!

I ended up removing quite a few unused files. Don't use 'em, don't need 'em.

### CODE REVIEW

File: code/modules/client/client procs.dm [hunk 1]
```diff
@@ -57,7 +57,6 @@
 	if( findtext(href,"<script",1,0) )
 		log_to_dd("Attempted use of scripts within a topic call, by [src]")
 		message_admins("Attempted use of scripts within a topic call, by [src]")
-		//del(usr)
 		return
```
Comment (Crazylemon64) [inline code/modules/client/client procs.dm]: It deleted the exploiter's client datum? kek

File: code/game/machinery/bees_apiary.dm [hunk 2]
```diff
@@ -1,245 +0,0 @@
-//http://www.youtube.com/watch?v=-1GadTfGFvU
-//i could have done these as just an ordinary plant, but fuck it - there would have been too much snowflake code
-
-/obj/machinery/apiary
```
Comment (Fox-McCloud) [inline code/game/machinery/bees_apiary.dm]: RIP in bees, Cael Code.

File: code/modules/client/client procs.dm (hunk 1 above)
Comment (Krausus) [inline code/modules/client/client procs.dm]: Nope. It deleted their mob. Honestly not sure what that was meant to accomplish.

File: code/modules/computer3/computers/cloning.dm [hunk 3]
```diff
@@ -1,363 +0,0 @@
-/obj/machinery/computer3/cloning
```
Comment (Crazylemon64) [inline code/modules/computer3/computers/cloning.dm]: Part of me is torn to see chunks of `computer3` bite the dust

File: code/modules/client/client procs.dm (hunk 1 above)
Comment (PJB3005) [inline code/modules/client/client procs.dm]: It's also hilariously easy to bypass this check.

**Pull request merged**